components/raftstore/src/store/txn_ext.rs

```diff
@@ -244,12 +244,20 @@ impl PeerPessimisticLocks {
         // Locks that are marked deleted still need to be moved to the new regions,
         // and the deleted mark should also be cleared.
         // Refer to the comment in `PeerPessimisticLocks` for details.
-        let removed_locks = self.map.drain_filter(|key, _| {
-            let key = &**key.as_encoded();
+        // There is no drain_filter for BtreeMap, so extra clone are needed.
+        let mut removed_locks = Vec::new();
+        self.map.retain(|key, value| {
+            let encoded_key = key.as_encoded().as_slice();
```

Comment:
`key` is internally in encoded form, and `as_encoded` gets an reference to it. It's not very necessary to emphasize it's encoded here, as all other keys being used here are encoded too.